addons/docs/src/lib/docgen/createPropDef.ts

```diff
@@ -4,31 +4,45 @@ import { TypeSystem, DocgenInfo, DocgenType } from './types';
 import { JsDocParsingResult } from '../jsdocParser';
 import { createDefaultValue } from './createDefaultValue';
 
-export type PropDefFactory = (
-  propName: string,
-  docgenInfo: DocgenInfo,
-  jsDocParsingResult?: JsDocParsingResult
-) => PropDef;
+export interface PropDefFactoryProps {
+  name: string;
+  defaultPropValue: any;
+  docgenInfo: DocgenInfo;
+  jsDocParsingResult?: JsDocParsingResult;
+}
+
+export type PropDefFactory = (props: PropDefFactoryProps) => PropDef;
+
+export interface CreatePropDefProp extends PropDefFactoryProps {
+  type: DocgenType;
+}
 
-function createBasicPropDef(name: string, type: DocgenType, docgenInfo: DocgenInfo): PropDef {
+function createBasicPropDef({
+  name,
+  defaultPropValue,
+  type,
+  docgenInfo,
+}: Omit<CreatePropDefProp, 'jsDocParsingResult'>): PropDef {
```

Comment:
Not sure about that coupling between the `createPropDef` and this function.

I understand it is convenient to re-use the interface but I think those are 2 separate things that should evoluate separately. The `Omit` you added prove it.